client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,448 @@
+		document
+			.getElementById( 'woocommerce-order-actions' )
+			.insertAdjacentHTML( 'afterend', shipmentTrackingHtml );
+
+		// Need to refresh so the new metaboxes are sortable.
+		window.jQuery( '#normal-sortables' ).sortable( 'refresh' );
+		window.jQuery( '#side-sortables' ).sortable( 'refresh' );
+
+		Promise.all( [
+			new Promise( ( resolve, reject ) => {
+				const script = document.createElement( 'script' );
+				script.src = js;
+				script.async = true;
+				script.onload = resolve;
+				script.onerror = reject;
+				document.body.appendChild( script );
+			} ),
+			new Promise( ( resolve, reject ) => {
+				const head = document.getElementsByTagName( 'head' )[ 0 ];
+				const link = document.createElement( 'link' );
+				link.rel = 'stylesheet';
+				link.type = 'text/css';
+				link.href = styles;
+				link.media = 'all';
+				link.onload = resolve;
+				link.onerror = reject;
+				head.appendChild( link );
+			} ),
+		] ).then( () => {
+			this.setState( {
+				wcsAssetsLoaded: true,
+				wcsAssetsLoading: false,
+				isShippingLabelButtonBusy: false,
+			} );
+			this.openWcsModal();
+		} );
+	}
+
+	getInstallText = () => {
+		const { activePlugins, wcsPluginSlug } = this.props;
+		if ( activePlugins.includes( wcsPluginSlug ) ) {
+			// If WCS is active, then the only remaining step is to agree to the ToS.
+			return __(
+				'You\'ve already installed WooCommerce Shipping. By clicking "Create shipping label", you agree to its {{tosLink}}Terms of Service{{/tosLink}}.',
+				'woocommerce-admin'
+			);
+		}
+		return __(
+			'By clicking "Create shipping label", {{wcsLink}}WooCommerce Shipping{{/wcsLink}} will be installed and you agree to its {{tosLink}}Terms of Service{{/tosLink}}.',
+			'woocommerce-admin'
+		);
+	};
+
+	openWcsModal() {
+		if ( window.wcsGetAppStore ) {
+			const wcsStore = window.wcsGetAppStore(
+				'wc-connect-create-shipping-label'
+			);
+			const state = wcsStore.getState();
+			const { orderId } = this.state;
+			const siteId = state.ui.selectedSiteId;
+
+			wcsStore.dispatch( {
+				type: 'WOOCOMMERCE_SERVICES_SHIPPING_LABEL_OPEN_PRINTING_FLOW',
+				orderId,
+				siteId,
+			} );
+			wcsStore.dispatch( {
+				type: 'NOTICE_CREATE',
+				notice: {
+					duration: 10000,
+					status: 'is-success',
+					text: __(
+						'Plugin installed and activated',
+						'woocommerce-admin'
+					),
+				},
+			} );
+			document.getElementById(
+				'woocommerce-admin-print-label'
+			).style.display = 'none';
+		}
+	}
+
+	render() {
+		const { isDismissModalOpen, showShippingBanner } = this.state;
+		if ( ! showShippingBanner ) {
+			return null;
+		}
+
+		return (
+			<div>
+				<div className="wc-admin-shipping-banner-container">
+					<img
+						className="wc-admin-shipping-banner-illustration"
+						src={ wcAdminAssetUrl + 'shippingillustration.svg' }
+						alt={ __( 'Shipping ', 'woocommerce-admin' ) }
+					/>
+					<Button
+						disabled={ this.state.isShippingLabelButtonBusy }
```

Comment:
Since we are already deconstructing `this.state` above, perhaps we could add in `isShippingLabelButtonBusy` and `isShippingLabelButtonBusy` as well to tidy things up here a bit?

Reply:
yes